expressions/methods.md

```diff
@@ -29,14 +29,16 @@ After the return value there's a `=>` and then finally the function body. The va
 
 If you want to exit a function early then use the `return` command. If the function has a return type then you need to provide a value to return. If the function does not have a return type then `return` should appear on its own, without a value.
 
-```pony
-class C
-  fun factorial(x: U32): U32 =>
-    if x == 0 then
-	  return 1
-	end
+Pony applies tail call optimization to allow a recursive implementation such as the following factorial function:
 
-	x * factorial(x - 1)
+```pony
+fun factorial(x: I32): I32 ? =>
```

Comment:
Sorry, I just checked it, the call is optimised away. But I wouldn't rely on that behaviour for all recursive functions. For example, in a recursive Fibonacci implementation such as the following, the recursive calls aren't eliminated.

```pony
fun fib(n: USize): USize =>
  if n < 2 then
    n
  else
    fib(n - 1) + fib(n - 2)
  end
```

AFAIK, tail call optimisation in LLVM is only guaranteed when the recursive call really is in tail position.